crowbar_framework/app/models/api/storage.rb

```diff
@@ -15,6 +15,32 @@
 #
 
 module Api
-  class Storages
+  class Storage < Tableless
+    def repocheck
+      {}.tap do |ret|
+        ret["ceph"] = {}
+        ceph_node_architectures = NodeObject.find(
+          "roles:ceph-mon AND ceph_config_environment:*"
```

Comment:
And to answer your initial question: you won't find all ceph nodes this way, just some of them. It'd be better to find all applied ceph proposals, and then look at the elements in there.